### DISCUSSION TITLE
Testers Wanted!

### DISCUSSION THREAD

**Discussion by nico-loeber** [php-perfect/ddev-intellij-plugin#21] [Announcements]
We have finally finished all the features we think are relevant for version 1.0 of our DDEV Integration Plugin for IntelliJ Platforms. :tada:

Now we need your feedback if all features behave as you expect or if there might be something we didn't think of!
If you are an experienced DDEV user working with PHPStorm or Webstorm, you are qualified to be an alpha tester.

## I would like to participate, what do I have to do?

First, leave us a comment in this discussion and let us know which IDE you are using and on which operating system. We would like to know how many people are testing and in which environments.

Install the latest versions of DDEV (>=1.19), your IDE (>=2022.1) and of course the DDEV Integration Plugin (>= 1.0.0-alpha2).

Refer to the [testing pre-release](https://github.com/php-perfect/ddev-intellij-plugin/blob/main/CONTRIBUTING.md#testing-of-pre-releases) documentation for installation instructions.

### Give us Feedback!
- What do you like most, what can we improve? [Open a Discussion](https://github.com/php-perfect/ddev-intellij-plugin/discussions/new)
- Do you expereince any errors or crashes? [Create an Issue](https://github.com/php-perfect/ddev-intellij-plugin/issues/new)
- Do you have ideas for additional features? [Tell Us](https://github.com/php-perfect/ddev-intellij-plugin/discussions/categories/ideas)

In case you already like what you see, give us a positive rating in the [Jetbrains Marketplace](https://plugins.jetbrains.com/plugin/18813-ddev-integration/reviews)! :)

Thank you!

**Comment by nico-loeber** [comment]
PHPStorm, Windows 11 (WSL)

**Comment by SpraxDev** [comment]
WebStorm, Debian 11

**Comment by juulliian** [comment]
PhpStorm, macOS 12.3, Apple silicon, Docker Desktop and Colima

**Comment by marcvangend** [comment]
PhpStorm, MacOS 11, Intel processor, Docker Desktop

**Comment by kgaut** [comment]
Phpstorm 2022.1, on kubuntu 21.10

**Comment by cmuench** [comment]
Tested on my Laptop (Manjaro Linux)

:heavy_check_mark:  Run DDEV actions via GUI
:heavy_check_mark:Quick access DDEV Services
:heavy_check_mark:Auto configure PHP Remote Interpreter
:x: Auto configure data source
:heavy_check_mark: Assistance and completion for DDEV configuration file
:question: DDEV version check (where can I test it?)
:heavy_check_mark: Integrated DDEV terminal

![image](https://user-images.githubusercontent.com/211294/163943758-00f4ca27-fa46-4a1a-813d-a9dfe81e0fcd.png)

**Reply by cmuench** [reply]
> Tested on my Laptop (Manjaro Linux)
> 
> heavy_check_mark Run DDEV actions via GUI heavy_check_markQuick access DDEV Services heavy_check_markAuto configure PHP Remote Interpreter x Auto configure data source heavy_check_mark Assistance and completion for DDEV configuration file question DDEV version check (where can I test it?) heavy_check_mark Integrated DDEV terminal
> 
> ![image](https://user-images.githubusercontent.com/211294/163943758-00f4ca27-fa46-4a1a-813d-a9dfe81e0fcd.png)

The port of the integration was detected. Found that I have to replace "localhost" with "127.0.0.1" to get it run.

jdbc:mysql://localhost:49171/db?user=db&password=db

vs. 

jdbc:mysql://127.0.0.1:49171/db?user=db&password=db

**Reply by cmuench** [reply]
Maybe it's an issue in Manjaro. A ping to localhost is resolved to a IPv6 address. Will try that on my Arch Linux Workstation, too.

**Reply by nico-loeber** [reply]
I've created an issue for that #24, please post any updates there.

**Reply by nico-loeber** [reply]
Ah, you asked for the version check. If a new DDEV version is published you should see a notification on the lower right hand side, notifying you about the update with a link to the DDEV installation / update instructions.

![image](https://user-images.githubusercontent.com/18308418/164055829-92258bd8-9e01-4665-81b6-3e0332aaa81c.png)

Automatic updates are planned but there are some difficulties regarding elevated rights and OS specific stuff.

You can manually trigger the version check by searching and running "Check for DDEV updates".
![image](https://user-images.githubusercontent.com/18308418/164056361-c2ed12c7-c931-457d-9abc-2abf83526415.png)

**Comment by calien666** [comment]
First test:
* PhpStorm 2022.1
* Ubuntu 2021.10 64bit
* docker native from docker repositories
* existing ddev project

Error in Automatic PHP Interpreter Configuration

After disabling in settings, ddev status is shown, starting of website and mailhog successful

ddev console accesses directly and is working.

Still looking for GUI actions, not found until now.

Wanted to test, if own written commands are usable, too

**Reply by cmuench** [reply]
GUI actions should be found here: Tools -> DDEV

**Reply by calien666** [reply]
Thanks, they are there. And Working.
But not the best place for getting my work more effective. terminal is always open and so the terminal and typing is faster than using the GUI

**Reply by nico-loeber** [reply]
Same for me @calien666, but in my company there are many people who prefer a GUI over a binary on the command line.
But even if you feel comfortable on the command line it might safe you some time in some situations, as you do not need to switch context. Especially if you use only one screen. In case you mean the integrated terminal, its use is complicated, at least on windows, as you do have other key bindings as you would expect.

You mentioned an error in the automatic interpreter configuration. Could you please open an issue and add some detail? Thanks!

**Comment by cmuench** [comment]
Arch Linux

:heavy_check_mark: Run DDEV actions via GUI
:heavy_check_mark: DDEV Services
:heavy_check_mark: Auto configure PHP Remote Interpreter
:heavy_check_mark: Auto configure data source
:heavy_check_mark: Assistance and completion for DDEV configuration file
:heavy_check_mark: Integrated DDEV terminal

**Comment by jonaseberle** [comment]
I'd like to test with PhpStorm 2022.1.1 (Manjaro and Ubuntu) when the plugin is compatible with that version (https://plugins.jetbrains.com/plugin/18813-ddev-integration).

**Reply by cmuench** [reply]
You can download the alpha 3 version here: https://github.com/php-perfect/ddev-intellij-plugin/releases/tag/1.0.0-alpha3

**Reply by nico-loeber** [reply]
Or the signed one (JetBrains and developer certificate) here https://plugins.jetbrains.com/plugin/18813-ddev-integration/versions.
If the marketplace says it's not compatible, it's lying! Probably because there is no STABLE version for your IDE. ;)

**Reply by nico-loeber** [reply]
And of course thanks for testing @jonaseberle!

**Reply by jonaseberle** [reply]
Thank you very much! I should have read more carefully that you provide [great instructions for testing](https://github.com/php-perfect/ddev-intellij-plugin/blob/main/CONTRIBUTING.md#testing-of-pre-releases).

I've tested on Manjaro:
✔️ Auto configure data source (<-- This is my favourite feature!)
✔️ Assistance and completion for DDEV configuration file
✔️ Integrated DDEV terminal
❎  Tools -> DDEV -> share: This opens in the integrated terminal but only outputs `Running /home/jonas/bin/ngrok http http://127.0.0.1:49214 -region eu ` (the `-region` is a custom `ngrok_args: -region eu` setting). Running it in my host terminal shows the expected ngrok status page. (In my opinion the internal terminal is just broken)

I haven't found the custom DDEV commands or services. I am happy to test if I find them but I guess I will not use them later on because I am more fluent with the terminal.

**Reply by nico-loeber** [reply]
Thanks for your feedback @jonaseberle!
Could you please open an issue for everything you think is broken, so I can track them better and have some more detail?

Check out the feature documentation to find the actions and services:
- https://github.com/php-perfect/ddev-intellij-plugin/wiki/Features#ddev-actions
- https://github.com/php-perfect/ddev-intellij-plugin/wiki/Features#quick-access-to-ddev-services

If you prefer a terminal, I think that at least the automatic configuration of the php interpreter and the data source are performance gainer, as you already mentioned. :)

**Reply by jonaseberle** [reply]
Thank you. 
Nevermind, I just misinterpreted what was meant by services/actions ;) I was expecting to see the custom services/commands.

✔️  Run DDEV default commands (start/restart/stop/poweroff) via GUI
✔️  Run DDEV launch commands (mailhog/website) from DDEV footer icon popup menu

For DDEV share I opened an issue: https://github.com/php-perfect/ddev-intellij-plugin/issues/53
